crates/sui-analytics-indexer/src/lib.rs

```diff
@@ -26,15 +59,269 @@ pub struct AnalyticsIndexerConfig {
+            FileType::MoveCall => Path::from(MOVE_CALL_PREFIX),
+        }
+    }
+    pub fn file_path(
+        &self,
+        file_format: FileFormat,
+        epoch_num: EpochId,
+        checkpoint_sequence_num: u64,
+    ) -> Path {
+        self.dir_prefix()
+            .child(format!("{}{}", EPOCH_DIR_PREFIX, epoch_num))
+            .child(format!(
+                "{}.{}",
+                checkpoint_sequence_num,
+                file_format.file_suffix()
+            ))
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct FileMetadata {
+    pub file_type: FileType,
+    pub file_format: FileFormat,
+    pub epoch_num: u64,
+    pub checkpoint_seq_range: Range<u64>,
+}
+
+impl FileMetadata {
+    fn new(
+        file_type: FileType,
+        file_format: FileFormat,
+        epoch_num: u64,
+        checkpoint_seq_range: Range<u64>,
+    ) -> FileMetadata {
+        FileMetadata {
+            file_type,
+            file_format,
+            epoch_num,
+            checkpoint_seq_range,
+        }
+    }
+    pub fn file_path(&self) -> Path {
+        self.file_type.file_path(
+            self.file_format,
+            self.epoch_num,
+            self.checkpoint_seq_range.start,
+        )
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct CheckpointUpdates {
+    files: Vec<FileMetadata>,
+    pub(crate) manifest: Manifest,
+}
+
+impl CheckpointUpdates {
+    pub fn new(
+        epoch_num: u64,
+        checkpoint_sequence_number: u64,
+        files: Vec<FileMetadata>,
+        manifest: &mut Manifest,
+    ) -> Self {
+        manifest.update(epoch_num, checkpoint_sequence_number, files.clone());
+        CheckpointUpdates {
+            files,
+            manifest: manifest.clone(),
+        }
+    }
+
+    pub fn new_for_epoch(
+        file_format: FileFormat,
+        epoch_num: u64,
+        checkpoint_range: Range<u64>,
+        manifest: &mut Manifest,
+    ) -> Self {
+        let files: Vec<_> = FileType::iter()
+            .map(|f| FileMetadata::new(f, file_format, epoch_num, checkpoint_range.clone()))
+            .collect();
+        CheckpointUpdates::new(epoch_num, checkpoint_range.end, files, manifest)
+    }
+
+    pub fn files(&self) -> Vec<FileMetadata> {
+        self.files.clone()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct ManifestV1 {
+    pub version: u8,
+    pub next_checkpoint_seq_num: u64,
+    pub file_metadata: Vec<FileMetadata>,
+    pub epoch: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub enum Manifest {
+    V1(ManifestV1),
+}
+
+impl Manifest {
+    pub fn new(epoch: u64, next_checkpoint_seq_num: u64) -> Self {
+        Manifest::V1(ManifestV1 {
+            version: 1,
+            next_checkpoint_seq_num,
+            file_metadata: vec![],
+            epoch,
+        })
+    }
+    pub fn files(&self) -> Vec<FileMetadata> {
```

Comment:
same all over here for the newlines, but really up to you

Reply:
added everywhere